test/unittest/CMakeLists.txt

```diff
@@ -60,25 +65,37 @@ foreach(NATIVE_TEST ${NATIVE_TESTS})
     endif()
 endforeach()
 
-if (CMAKE_BUILD_TYPE STREQUAL "Coverage")
-    enable_coverage_report(COVERAGE_REPORT_FORMATS html sonarqube
-                           ROOT_DIRECTORY ${LIBCYPHAL_ROOT}
-                           OUT_REPORT_INDICIES LOCAL_COVERAGE_REPORT_INDICIES
-    )
-endif()
-
 add_custom_target(
-     build_unittests
-     DEPENDS
-          ${ALL_TESTS_BUILD}
+    build_unittests
+    DEPENDS
+        ${ALL_TESTS_BUILD}
 )
 
 add_custom_target(
-     run_unittests
-     DEPENDS
-          ${ALL_TESTS_RUN}
+    run_unittests
+    DEPENDS
+        ${ALL_TESTS_RUN}
 )
 
 set_directory_properties(PROPERTIES
     IN_BUILD_TESTS "${ALL_TESTS_RUN}"
 )
+
+if (CMAKE_BUILD_TYPE STREQUAL "Coverage")
+
+    enable_coverage_report(COVERAGE_REPORT_FORMATS html sonarqube
+        ROOT_DIRECTORY ${LIBCYPHAL_ROOT}
+        OUT_REPORT_INDICIES LOCAL_COVERAGE_REPORT_INDICIES
+    )
+
+    # we need to build at least one source file to get sonarqube to look at the cetl
```

Comment:
why cetl

Reply:
You may want to know that there's another way to use Sonar with a header-only library: just analyze your test build, and configure exclusion rules to exclude all .cpp files, as done here: https://github.com/Zubax/dyshlo/blob/27e0f89970e476073eef9c9a556d74a19c632da2/tools/run_sonar.sh#L59-L74